airflow/models/dag.py

```diff
@@ -347,6 +347,9 @@ def __init__(
     ):
         from airflow.utils.task_group import TaskGroup
 
+        if tags and any(len(tag) > 100 for tag in tags):
```

Comment:
I just wonder - why 100 ? Do we know why the 100 causes the DAGs to disappear (in the original issue? ) I think having such a magic constant here is not good - it needs to come from somewhere ?